docs/content/basic/authenticating-oauth.md

```diff
@@ -4,256 +4,703 @@ lang: en
-        failure: (error, installOptions , req, res) => {
-          // Do custom failure logic here
-          res.send('failure');
-        }
-      },
-      stateStore: {
-        // Do not need to provide a `stateSecret` when passing in a stateStore
-        // generateStateParam's first argument is the entire InstallUrlOptions object which was passed into generateInstallUrl method
-        // the second argument is a date object
-        // the method is expected to return a string representing the state
-        generateStateParam: async (installUrlOptions, date) => {
-          // generate a random string to use as state in the URL
-          const randomState = randomStringGenerator();
-          // save installOptions to cache/db
-          await myDB.set(randomState, installUrlOptions);
-          // return a state string that references saved options in DB
-          return randomState;
-        },
-        // verifyStateParam's first argument is a date object and the second argument is a string representing the state
-        // verifyStateParam is expected to return an object representing installUrlOptions
-        verifyStateParam:  async (date, state) => {
-          // fetch saved installOptions from DB using state reference
-          const installUrlOptions = await myDB.get(randomState);
-          return installUrlOptions;
-        }
-      },
-  }
-});
+const database = {
+  store: {},
+  async get(key) {
+    return this.store[key];
+  },
+  async delete(key) {
+    delete this.store[key];
+  },
+  async set(key, value) {
+    this.store[key] = value;
+  },
+};
 ```
-</details>
\ No newline at end of file
+
+</details>
+
+### Sign in with Slack
+
+Right now Bolt does not support [Sign in with Slack][siws] out-of-the-box. This
+still continues to remain an option using APIs from the
+[`@slack/web-api`][web-api] package, which aim to make implementing OpenID
+Connect (OIDC) connections simple. Alternative [routes][custom-routes] might be
+required.
+
+Explore [this relevant package documentation][oidc] for reference and example.
+
+### Extra authorizations
+
+If you need additional authorizations, such as user tokens from users of a team
+where your app is already installed, or have a reason to dynamically generate an
+install URL, an additional installation is required.
+
+Generating a new installation URL requires a few steps: manually instantiate an
+`ExpressReceiver`, assign the instance to a variable named `receiver`, and then
+call `receiver.installer.generateInstallUrl()`.
+
+Read more about `generateInstallUrl()` in the
+[OAuth docs][generate-install-url].
+
+### Common errors
+
+Occasional mishaps in various places throughout the OAuth process can cause
+errors, but these often have meaning! Explore [the API documentation][errors]
+for additional details for common error codes.
+
+[add-to-slack]: https://github.com/slackapi/node-slack-sdk/blob/main/packages/oauth/src/default-render-html-for-install-path.ts
```

Comment:
i had no idea you could do this! 🥳

Reply:
It is such an important feature. Blessed markdown 🕴️